bullet_train-outgoing_webhooks/test/models/concerns/webhooks/outgoing/endpoint_deactivatable_test.rb

```diff
@@ -0,0 +1,281 @@
+require "test_helper"
+require "minitest/mock"
+
+class Webhooks::Outgoing::EndpointDeactivatableTest < ActiveSupport::TestCase
+  include ActiveJob::TestHelper
+  include CreateWebhooksTestHelpers
+
+  def set_config(config)
+    BulletTrain::OutgoingWebhooks::Engine.config.define_singleton_method(:outgoing_webhooks) { config }
+  end
+
+  def test_enabled_config
+    {
+      automatic_endpoint_deactivation_enabled: true,
+      automatic_endpoint_deactivation_settings: {
+        max_limit: 5, # lowered for testing
+        deactivation_in: 3.days,
+      }
+    }
+  end
+
+  def test_disabled_config
+    {automatic_endpoint_deactivation_enabled: false}
+  end
+
+  setup do
+    @team = Team.create!(name: "test-team")
+  end
+
+  test "#deactivation_processing does nothing when feature is disabled" do
+    set_config(test_disabled_config)
+    endpoint = create_endpoint
+
+    endpoint.stub(:deactivate!, -> { raise "deactivate! should not be called" }) do
+      endpoint.stub(:mark_for_deactivation!, -> { raise "mark_for_deactivation! should not be called" }) do
+        assert_nil endpoint.deactivation_processing
+      end
+    end
+  end
+
+  test "#deactivation_processing does nothing when endpoint is already deactivated" do
+    set_config(test_enabled_config)
+    endpoint = create_endpoint(deactivated_at: 1.hour.ago)
+
+    assert endpoint.deactivated?
+
+    endpoint.stub(:deactivate!, -> { raise "deactivate! should not be called" }) do
+      endpoint.stub(:mark_for_deactivation!, -> { raise "mark_for_deactivation! should not be called" }) do
+        endpoint.stub(:increment!, -> { raise "increment! should not be called" }) do
+          assert_nil endpoint.deactivation_processing
+        end
+      end
+    end
+  end
+
+  test "#deactivation_processing deactivates endpoint when deactivation_in period has passed" do
+    set_config(test_enabled_config)
+    endpoint = create_endpoint(deactivation_limit_reached_at: 4.days.ago)
+
+    assert endpoint.should_be_deactivated?
+    assert_not endpoint.deactivated?
+    assert_equal 0, endpoint.consecutive_failed_deliveries
```

Comment:
This assertion is a little bit confusing. If an endpoint ever got to the point where `endpoint.should_be_deactivated?` is `true` shouldn't `endpoint.consecutive_failed_deliveries` be greater than or equal to the `max_limit`? I don't think it really affects the logical flow of the code or the validity of the test, but as I was reading this test I started to think that `max_limit` and `deactivation_in` are for handling separate scenarios instead them collaborating on a single scenario. I guess what I'm saying is that it doesn't seem like this matches a real world scenario. Or am I misunderstanding the deactivation flow?